core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandler.java

```diff
@@ -368,7 +371,8 @@ private void setFinalResult(
         logServerWarnings(callback.statement, executionProfile, executionInfo.getWarnings());
       }
     } catch (Throwable error) {
-      setFinalError(callback.statement, error, callback.node, -1);
+      ExecutionInfo executionInfo = defaultExecutionInfo(callback, -1).build();
```

Comment:
I know that it worked this way before, but it seems odd to me that we would use `-1` for the execution here instead of `callback.execution`.   I guess it's good to leave it this way for consistency like you did, but maybe idea here is "Something unpredictable unexpected happened here that we can't blame on the request itself.